src/pages/RepoPage/SettingsTab/SettingsTab.spec.tsx

```diff
@@ -0,0 +1,106 @@
+import { render, screen } from '@testing-library/react'
+import { MemoryRouter, Route } from 'react-router-dom'
+
+import { useOwner } from 'services/user'
+import { useFlags } from 'shared/featureFlags'
+
+import SettingsTab from './SettingsTab'
+
+jest.mock('services/user')
+const mockedUseOwner = useOwner as jest.Mock
+jest.mock('shared/featureFlags')
+const mockedUseFlags = useFlags as jest.Mock
+
+jest.mock('./tabs/GeneralTab', () => () => 'General Tab')
+
+const wrapper: (initialEntries?: string) => React.FC<React.PropsWithChildren> =
+  (initialEntries = '/gh/codecov/codecov-client/settings') =>
+  ({ children }) =>
+    (
+      <MemoryRouter initialEntries={[initialEntries]}>
+        <Route path="/:provider/:owner/:repo/settings">{children}</Route>
+      </MemoryRouter>
+    )
+
+interface SetupArgs {
+  isCurrentUserPartOfOrg?: boolean
+}
+
+describe('SettingsTab', () => {
+  function setup({ isCurrentUserPartOfOrg = true }: SetupArgs) {
+    mockedUseOwner.mockReturnValue({ data: { isCurrentUserPartOfOrg } })
+    mockedUseFlags.mockReturnValue({ inAppMarketingTab: true })
+  }
+
+  describe('Render for a repo', () => {
+    it('renders the right links', async () => {
+      setup({})
+      render(<SettingsTab />, { wrapper: wrapper() })
+
+      expect(
+        await screen.findByRole('link', { name: /General/ })
+      ).toBeInTheDocument()
```

Comment:
Not gonna ask for any changes, but moving forward can we pull these `*By*` out into their own vars and than use the vars in the expects